samples: net: lwm2m: Add support for nRF52840 on nRF9160DK

SeppoTakalo opened the pull request:
Add support for running LwM2M client on nRF52840 chip using nRF9160 as a modem through serial port.
nRF9160 should run Serial LTE Modem application build from nRF Connect SDK.

nRF9160DK board has two MCU, nRF9160 and nRF52840.

This is easier to demonstrate the modem, instead of running just `samples/net/cellular_modem`
It shows better how the modem is integrated into Zephyr network APIs and do not necessary require any changes to application or protocol stack.

SeppoTakalo committed:
samples: net: lwm2m: Add support for nRF52840 on nRF9160DK

Add support for running LwM2M client on nRF52840 chip
using nRF9160 as a modem through serial port.
nRF9160 should run Serial LTE Modem application build
from nRF Connect SDK.

nRF9160DK board has two MCU, nRF9160 and nRF52840.

Signed-off-by: Seppo Takalo <seppo.takalo@nordicsemi.no>

jukkar commented on samples/net/lwm2m_client/Kconfig:
```diff
@@ -27,10 +27,14 @@ config LWM2M_APP_TLS_TAG
 	  TLS security tag to use when connecting.
 endif
 
+config HAS_INTERNET
```
This sounds a bit optimistic :smile: should we call this `HAS_CONNECTIVITY` instead.

It is unfortunate that this sample did not use the `NET_SAMPLE` prefix for Kconfig symbols like what is done in many other networking samples. Now `LWM2M_` prefix is used  which gives the impression that the option is part of LWM2M library which this sample is not.

SeppoTakalo replied:
Well... those emulated platforms have connectivity too.
I tried to somehow separate the defaults that might actually have global internet connectivity and use public Leshan address.


I see, the `LWM2M_APP` is probably not the best.. maybe I should refactor to `NET_SAMPLE_LWM2M` instead.

SeppoTakalo replied:
`config NET_SAMPLE_MIGHT_HAVE_INTERNET`

SeppoTakalo committed:
samples: net: lwm2m: Refactor to use NET_SAMPLE Kconfig prefix

It is easier to separate a sample application Kconfig
values from ones that configure something inside Zephyr.

Signed-off-by: Seppo Takalo <seppo.takalo@nordicsemi.no>

SeppoTakalo commented:
As suggested by @jukkar I refactored all sample application specific Kconfigs to use `NET_SAMPLE_LWM2M` prefix so it is easier to see which ones configure the system, and which ones belong to the sample.
